pkg/kubectl/cmd/set/set_env.go

```diff
@@ -175,18 +181,20 @@ func keyToEnvName(key string) string {
 	return strings.ToUpper(validEnvNameRegexp.ReplaceAllString(key, "_"))
 }
 
-func (o *EnvOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
-	if o.All && len(o.Selector) > 0 {
-		return fmt.Errorf("cannot set --all and --selector at the same time")
-	}
-	resources, envArgs, ok := envutil.SplitEnvironmentFromResources(args)
-	if !ok {
-		return cmdutil.UsageErrorf(o.Cmd, "all resources must be specified before environment changes: %s", strings.Join(args, " "))
-	}
-	if len(o.Filenames) == 0 && len(resources) < 1 {
-		return cmdutil.UsageErrorf(cmd, "one or more resources must be specified as <resource> <name> or <resource>/<name>")
+func (o *EnvOptions) addEnvVarToList(key string, env []v1.EnvVar, envVar v1.EnvVar) []v1.EnvVar {
+	if len(o.Keys) > 0 {
```

Comment:
Seems weird to have the skip here.  Why not skip down in the method iterating over keys?  This feels like a function that sometimes disobeys me.

A `contains(needle string, haystack []string)` could hold this logic.

Reply:
Originally had the code where we iterate over the keys, but was asked to move it to a helper since it was repeated for both configmap and secret